zsond.go

```diff
@@ -67,37 +72,30 @@ func handle(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusForbidden)
 		return
 	}
-	size := 128 * 1024
-	reader := bufio.NewReaderSize(r.Body, size)
-	buf, err := reader.Peek(size)
-	if err != nil && err != io.EOF {
-		http.Error(w, err.Error(), http.StatusForbidden)
-		return
-	}
-	if len(buf) == 0 {
-		http.Error(w, "empty body", http.StatusForbidden)
-		return
-	}
-	logType := findPath(buf)
-	if logType == "" {
-		http.Error(w, "no #path directive found", http.StatusForbidden)
+
+	reader := bufio.NewReader(r.Body)
+	header, path, err := readPath(reader)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	file, name, err := openfile(dirPath, logType)
+	file, _, err := openfile(dirPath, path)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
+		http.Error(w, err.Error(), http.StatusServiceUnavailable)
 		return
 	}
+	if _, err := file.Write(header); err != nil {
+		http.Error(w, err.Error(), http.StatusServiceUnavailable)
```

Comment:
A write error definitely seems like StatusInternalServerError material.

```suggestion
		http.Error(w, err.Error(), http.StatusInternalServerError)
```